### PR TITLE
added ui and implementation for functionality of set parking lots

### PR DESCRIPTION
**Pull request by LucreciaLucero**
Added the ui and implementation of functionality for set parking lots
.
**Structure folder and Test**: 
image
image
image

Video - Pixel 3a:
https://user-images.githubusercontent.com/84336476/127542619-be45d9b9-0a24-447f-8b6d-b998b4f9d4cc.mp4

Video - Nexus 5X:
https://user-images.githubusercontent.com/84336476/127543556-3a4a25c0-be6b-4ce3-8d32-9f4aeb10c1dd.mp4

**Screen- Device**: Nexus 5X 
image
image
image

**Screen-Device**: Pixel 3a
Screenshot_1627581052
Screenshot_1627581141
Screenshot_1627581155

### CODE REVIEW

File: app/src/main/java/com/example/parkingsystemkotlin/mvp/presenter/DialogFragmentParkingLotsPresenter.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+class DialogFragmentParkingLotsPresenter(private var view: DialogFragmentParkingLotsContract.ViewContract) :
+    DialogFragmentParkingLotsContract.PresenterContract {
+
+    override fun onSaveButtonPressed(listenerParkingLots: DialogFragmentParkingLotsListener) {
+        val spaces: String = view.getParkingsAvailable()
+        if (spaces.isNotEmpty()) {
+            val numberSpaces = spaces.toInt()
+            listenerParkingLots.listenerSetParkingLotsOnClick(numberSpaces)
+        }
```
Comment (nicolbidone) [inline app/src/main/java/com/example/parkingsystemkotlin/mvp/presenter/DialogFragmentParkingLotsPresenter.kt:12]: please avoid creating single use vars

File: app/src/main/java/com/example/parkingsystemkotlin/mvp/view/DialogFragmentParkingLotsView.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import com.example.parkingsystemkotlin.databinding.DialogFragmentParkingLotsBinding
+import com.example.parkingsystemkotlin.fragment.DialogFragmentParkingLots
+import com.example.parkingsystemkotlin.mvp.contract.DialogFragmentParkingLotsContract
+import com.example.parkingsystemkotlin.mvp.view.base.FragmentView
+
+class DialogFragmentParkingLotsView(
+    private var binding: DialogFragmentParkingLotsBinding,
+    dialogFragmentParkingLots: DialogFragmentParkingLots
+) :
+    FragmentView(dialogFragmentParkingLots), DialogFragmentParkingLotsContract.ViewContract {
```
Comment (nicolbidone) [inline app/src/main/java/com/example/parkingsystemkotlin/mvp/view/DialogFragmentParkingLotsView.kt]: this should be ) : FragmentView ...

File: app/src/main/java/com/example/parkingsystemkotlin/mvp/view/DialogFragmentParkingLotsView.kt [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    override fun getParkingsAvailable(): String = binding.inputTextDialogFragmentParkingLots.text.toString()
+
+    override fun dismissDialogFragmentParkingLots() {
+        fragment?.let {
+            val dialogFragmentRef: DialogFragmentParkingLots = it as DialogFragmentParkingLots
+            dialogFragmentRef.dismiss()
```
Comment (nicolbidone) [inline app/src/main/java/com/example/parkingsystemkotlin/mvp/view/DialogFragmentParkingLotsView.kt]: please avoid creating single use vars

File: app/src/main/java/com/example/parkingsystemkotlin/mvp/view/ParkingView.kt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import com.example.parkingsystemkotlin.mvp.view.base.ActivityView
+import com.example.parkingsystemkotlin.utils.ConstantUtils
+import com.example.parkingsystemkotlin.utils.toast
+
+class ParkingView(activity: Activity) : ActivityView(activity), ParkingContract.ViewContract {
+
+    override fun showDialogParkingsAvailable() {
+        fragmentManager?.let {
+            val dialog: DialogFragment = DialogFragmentParkingLots()
+            dialog.show(it, ConstantUtils.DIALOG_FRAGMENT_TAG)
```
Comment (nicolbidone) [inline app/src/main/java/com/example/parkingsystemkotlin/mvp/view/ParkingView.kt]: please avoid creating single use vars

File: app/src/main/res/layout/activity_main.xml [hunk 5]
```diff
@@ -1,18 +1,22 @@
...
-        app:layout_constraintTop_toTopOf="parent" />
+        android:layout_height="wrap_content" />
 
-</androidx.constraintlayout.widget.ConstraintLayout>
+    <Button
+        android:id="@+id/button_main_parking_size"
+        style="@style/MainButtonStyle"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content" />
+</LinearLayout>
```
Comment (nicolbidone) [inline app/src/main/res/layout/activity_main.xml]: Please use just Constraint Layout for this exercise

File: app/src/main/res/layout/dialog_fragment_parking_lots.xml [hunk 6]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (nicolbidone) [inline app/src/main/res/layout/dialog_fragment_parking_lots.xml]: Please use just Constraint Layout for this exercise

File: app/src/main/java/com/example/parkingsystemkotlin/mvp/view/base/ActivityView.kt [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+import android.app.Activity
+import android.content.Context
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.FragmentManager
+import java.lang.ref.WeakReference
+
+open class ActivityView(activity: Activity) {
+    private var activityRef: WeakReference<Activity> = WeakReference(activity)
+
+    private val activity: Activity?
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystemkotlin/mvp/view/base/ActivityView.kt]: this should be public, nos private

File: app/src/main/java/com/example/parkingsystemkotlin/mvp/view/base/ActivityView.kt [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+        get() = activityRef.get()
+
+    val context: Context?
+        get() = activity
```
Comment (SartiMau) [inline app/src/main/java/com/example/parkingsystemkotlin/mvp/view/base/ActivityView.kt:16]: this should be `activity?.context`

File: app/src/main/res/drawable-v24/border.xml [hunk 9]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (SartiMau) [inline app/src/main/res/drawable-v24/border.xml:1]: please format the code properly

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/res/drawable-v24/border.xml (hunk 9 above)
Comment (LucreciaLucero) [reply app/src/main/res/drawable-v24/border.xml:1]: done

File: app/src/main/java/com/example/parkingsystemkotlin/mvp/presenter/DialogFragmentParkingLotsPresenter.kt (hunk 1 above)
Comment (LucreciaLucero) [reply app/src/main/java/com/example/parkingsystemkotlin/mvp/presenter/DialogFragmentParkingLotsPresenter.kt:12]: done

File: app/src/main/java/com/example/parkingsystemkotlin/mvp/view/DialogFragmentParkingLotsView.kt (hunk 2 above)
Comment (LucreciaLucero) [reply app/src/main/java/com/example/parkingsystemkotlin/mvp/view/DialogFragmentParkingLotsView.kt]: done

File: app/src/main/res/layout/activity_main.xml (hunk 5 above)
Comment (LucreciaLucero) [reply app/src/main/res/layout/activity_main.xml]: done

File: app/src/main/java/com/example/parkingsystemkotlin/mvp/view/ParkingView.kt (hunk 4 above)
Comment (LucreciaLucero) [reply app/src/main/java/com/example/parkingsystemkotlin/mvp/view/ParkingView.kt]: done

File: app/src/main/java/com/example/parkingsystemkotlin/mvp/view/DialogFragmentParkingLotsView.kt (hunk 3 above)
Comment (LucreciaLucero) [reply app/src/main/java/com/example/parkingsystemkotlin/mvp/view/DialogFragmentParkingLotsView.kt]: done

File: app/src/main/res/layout/dialog_fragment_parking_lots.xml (hunk 6 above)
Comment (LucreciaLucero) [reply app/src/main/res/layout/dialog_fragment_parking_lots.xml]: done

File: app/src/main/java/com/example/parkingsystemkotlin/mvp/view/base/ActivityView.kt (hunk 8 above)
Comment (LucreciaLucero) [reply app/src/main/java/com/example/parkingsystemkotlin/mvp/view/base/ActivityView.kt:16]: we will dismiss this comment based on what we just talked

File: app/src/main/java/com/example/parkingsystemkotlin/mvp/view/base/ActivityView.kt (hunk 7 above)
Comment (LucreciaLucero) [reply app/src/main/java/com/example/parkingsystemkotlin/mvp/view/base/ActivityView.kt]: done

File: app/src/main/res/layout/activity_main.xml [hunk 10]
```diff
@@ -4,15 +4,29 @@
...
+    <Button
+        android:id="@+id/button_main_parking_size"
+        style="@style/MainButtonStyle"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintVertical_bias="0.453" />
```
Comment (SartiMau) [inline app/src/main/res/layout/activity_main.xml]: do you really need this vertical bias?

**Review by SartiMau — APPROVED** [review]

File: app/src/main/res/layout/activity_main.xml (hunk 10 above)
Comment (LucreciaLucero) [reply app/src/main/res/layout/activity_main.xml]: No sorry, I already deleted it

**Review by nicolbidone — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Pull request merged**